### DISCUSSION TITLE
GitHub App authentication as alternative to PATs

### DISCUSSION THREAD

**Discussion by Hdom** [cloudbase/garm#212] [Q&A] [answered]
Are there any plans for supporting GitHub app authentication as an alternative to the PATs? 

Outside of the security improvements related to github apps, PATs share the hourly rate limit of the account under which they are created, by default 5000, while every github app gets its own rate limit allocation of 15000 requests per hour.

**Comment by gabriel-samfira** [comment]
I don't see why not. Will look into this for the next release.

In the meantime, if you're hitting rate limits, try upgrading to [v0.1.4](https://github.com/cloudbase/garm/releases/tag/v0.1.4). It should alleviate some issues with many requests.

Also, from a security perspective, version v0.1.4 is also capable of using JIT runners. It should do so by default unless explicitly disabled.

**Reply by gabriel-samfira** [reply, marked as answer]
I added a new issue for this feature here: https://github.com/cloudbase/garm/issues/214

**Reply by gabriel-samfira** [reply]
This was added in https://github.com/cloudbase/garm/pull/228

**Reply by Hdom** [reply]
Very nice, I havent had the chance to implement it yet but I will work on it, thank you.

**Reply by Hdom** [reply]
@gabriel-samfira just following up on this, we switched over to github app authentication about a week ago and everything seems to be working well. Closing this discussion for now, thank you.
